### ISSUE TITLE
Inpainting for images with local information

### ISSUE THREAD

**Issue by spillerrec** [deeppomf/DeepCreamPy#16]
I have had my fun with demosaicing with a multi image approach by gathering information from multiple images (i.e. video) rather than ML. It is limited in scope, but when applicable produces great results and is guaranteed to match the ground truth for the pixels it can infer. More details here for the curious: https://spillerrec.dk/2015/07/removing-mosaic-censor/

The problem is that it cannot infer all the pixels and I need some way of inpainting the rest. The inpainting methods I tried so far doesn't work well with lines however. DCP, while initial sounding promising, doesn't seem to work on these kind of images at all:
![chris](https://user-images.githubusercontent.com/2292170/47384780-8bcd5500-d708-11e8-888b-18b92adcf643.png)
The result looks like this:
![chris](https://user-images.githubusercontent.com/2292170/47384988-1dd55d80-d709-11e8-8acb-4ca50dfad1cf.png)
I assume it needs some more information around each decensored area.

While I understand this probably is out of scope of this project, do you have any suggestions for any other approaches that could work well?

**Comment by deeppomf** [comment]
DCP was trained specifically for inpainting censor bars, heart-shaped censors, etc for hentai.
If you trained a neural network specifically for inpainting masks with patterns like the one in the above image for anime images, I think you could see some success.

**Comment by DonaldTsang** [comment]
@deeppomf that is a new feature you can try and tackle

**Comment by spillerrec** [comment]
That sounds promising, I might give that a try once I have some time.

@DonaldTsang I think it would be more useful to work on models for screentones and monochrome images. My project is too limited in scope and existing inpainting methods can usually be used with okay results.
